stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -969,7 +969,8 @@ for (hermtype, wrapper) in [(:Symmetric, :Symmetric), (:SymTridiagonal, :Symmetr
         function sincos(A::$hermtype{<:Real})
             n = checksquare(A)
             F = eigen(A)
-            S, C = Diagonal(similar(A, (n,))), Diagonal(similar(A, (n,)))
+            T = float(eltype(F.values))
```

Comment:
Not that it matters much, but I'm pretty sure that for `eigen` the eltype is already `float`ed. So, the `float` call is a noop in all cases, and could be left out. Same below.

Reply:
Let's leave it in, just in case some custom array doesn't convert its eigenvalues to `float`